test/Microsoft.ML.Tests/SvmLightTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.IO;
+using Microsoft.ML.Data;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Transforms;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public sealed class SvmLightTests : TestDataPipeBase
+    {
+        public SvmLightTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+#pragma warning disable 0649 // Disable warnings about unused members. They are used through reflection.
+        private sealed class SvmLightOutput
+        {
+            public float Label;
+            public float Weight;
+            [KeyType(ulong.MaxValue - 1)]
+            public ulong GroupId = ulong.MaxValue;
+            public ReadOnlyMemory<char> Comment;
+            public VBuffer<float> Features;
+        }
+#pragma warning restore 0649
+
+        [Fact]
+        public void TestSvmLightLoaderAndSaver()
+        {
```

Comment:
It appears that this function is encapsulating several tests. Is it possible to refactor them into different tests?

Reply:
Good suggestion, I'll do that in the next iteration.

---
In reply to: [332204210](https://github.com/dotnet/machinelearning/pull/4190#discussion_r332204210) [](ancestors = 332204210)